### PR TITLE
User Guide Rewrite

### PR DESCRIPTION
**Pull request by bryevdv**
Deploying these periodically to:

http://bokeh.pydata.org/en/test/docs/user_guide.html

### CODE REVIEW

File: sphinx/source/docs/tutorial/setup.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    :depth: 2
+
+This first section will guide you through quick installation instructions,
+and provide some short "test" code you can run to verify that your Bokeh
+installation is working as expected.
+
+Installing the Bokeh Library
+----------------------------
+
+The easiest way to install Boeh and all of its necessary dependencies is
```
Comment (fpliger) [inline sphinx/source/docs/tutorial/setup.rst]: typo: Boeh -> Bokeh

**Comment by ericdill** [comment]
@bryevdv , the link in your original post (http://bokeh.pydata.org/en/pydfw/docs/tutorial/plotting.html) results in a 404.  That user guide link (http://bokeh.pydata.org/en/test/docs/user_guide.html) you posted in #2145 works great

**Comment by bryevdv** [comment]
@ericdill thanks I updated it!

**Comment by bryevdv** [comment]
OK, this PR has gotten quite large. Going to merge this (since it is already better than the old UG) and add more in follow on PRs.

**Pull request merged**